app/src/main/kotlin/com/simplemobiletools/calculator/helpers/CalculatorImpl.kt

```diff
@@ -3,214 +3,121 @@ package com.simplemobiletools.calculator.helpers
 import android.content.Context
 import com.simplemobiletools.calculator.R
 import com.simplemobiletools.calculator.operation.OperationFactory
+import com.simplemobiletools.calculator.operation.base.UnaryOperation
 
 class CalculatorImpl(calculator: Calculator, val context: Context) {
-    var displayedNumber: String? = null
-    var displayedFormula: String? = null
-    var lastKey: String? = null
-    private var mLastOperation: String? = null
     private var mCallback: Calculator? = calculator
 
-    private var mIsFirstOperation = false
-    private var mResetValue = false
-    private var mBaseValue = 0.0
-    private var mSecondValue = 0.0
+    private var firstNumber: Double = 0.0
+    private var secondNumber: Double = 0.0
+    private var operator: String? = ""
+    private var decimalClicked: Boolean = false
+    private var decimalCounter = 0
+    private var secondNumberSet: Boolean = false
+    private var digits = 0
+
+    // used for testing
+    var displayedNumber: String? = null
+    var displayedFormula: String? = null
```

Comment:
From what I can tell displayedNumber and displayedForula are useless. If anything needs to know what is currently being displayed it should use R.id.result (grabbing the value from the view ) to be more robust rather than setting it ourselves.

Reply:
fixed

Reply:
Excellent, Moses protect you